proto/regen/ecocredit/v1alpha1/types.proto

```diff
@@ -117,4 +117,62 @@ message CreditTypeSeq {
 
   // The sequence number of classes of the credit type
   uint64 seq_number = 2;
+}
+
+message Filter {
+  oneof sum {
+    And and = 1;
+    Or or = 2;
```

Comment:
do we actually need `And` here? i cant really see how that would work in this context

Reply:
`And` is needed to combine multiple options (for instance class + data range) within a single filter.

Reply:
spoke with cory and i think i just misunderstood the design, hence my additional [curly suggestion](https://github.com/regen-network/regen-ledger/pull/610/files/318d9d97d77a069e82e74298e0643c61d5ad8bcb#r740417038), and this [issue](https://github.com/regen-network/regen-ledger/pull/610/files/318d9d97d77a069e82e74298e0643c61d5ad8bcb#r741916566).